js/src/index.ts

```diff
@@ -50,6 +54,46 @@ export class Schematic {
     this.apiKey = apiKey;
     this.eventQueue = [];
     this.useWebSocket = options?.useWebSocket ?? false;
+    this.debugEnabled = options?.debug ?? false;
+    this.offlineEnabled = options?.offline ?? false;
+
+    // Check for debug mode in URL query parameter
+    if (
+      typeof window !== "undefined" &&
+      typeof window.location !== "undefined"
+    ) {
+      const params = new URLSearchParams(window.location.search);
+
+      // Debug mode
+      const debugParam = params.get("schematic_debug");
+      if (
+        debugParam !== null &&
+        (debugParam === "" || debugParam === "true" || debugParam === "1")
+      ) {
+        this.debugEnabled = true;
+      }
+
+      // Offline mode
+      const offlineParam = params.get("schematic_offline");
+      if (
+        offlineParam !== null &&
+        (offlineParam === "" || offlineParam === "true" || offlineParam === "1")
+      ) {
+        this.offlineEnabled = true;
+        // When offline mode is enabled via URL, also enable debug mode to log events
+        this.debugEnabled = true;
+      }
+    }
+
+    // When offline mode is enabled via options, also enable debug logging if not explicitly disabled
+    if (this.offlineEnabled && options?.debug !== false) {
```

Comment:
oh ok. so you can do `{ offline: true, debug: false }` I guess that should be good